src/main/kotlin/org/jetbrains/research/testspark/helpers/psiHelpers/JavaPsiHelper.kt

```diff
@@ -173,145 +136,313 @@ class JavaPsiHelper : PsiHelperInterface {
+        val interestingMethods = mutableSetOf(psiMethod as JavaPsiMethodWrapper)
+        for (currentPsiMethod in allMethods) {
+            if ((currentPsiMethod as JavaPsiMethodWrapper).isConstructor) interestingMethods.add(currentPsiMethod)
+        }
+        val interestingPsiClasses = mutableSetOf(this)
+        interestingMethods.forEach { methodIt ->
+            methodIt.parameterList.parameters.forEach { paramIt ->
+                PsiTypesUtil.getPsiClass(paramIt.type)?.let { typeIt ->
+                    JavaPsiClassWrapper(typeIt).let {
+                        if (!it.qualifiedName.startsWith("java.")) {
+                            interestingPsiClasses.add(it)
+                        }
+                    }
+                }
+            }
+        }
+
+        return interestingPsiClasses.toMutableSet()
+    }
+
+    val isInterface: Boolean get() = psiClass.isInterface
+
+    val isAbstractClass: Boolean
+        get() {
+            psiClass.containingFile.virtualFile
+            if (psiClass.isInterface) return false
+
+            val methods = PsiTreeUtil.findChildrenOfType(psiClass, PsiMethod::class.java)
+            for (psiMethod: PsiMethod in methods) {
+                if (psiMethod.body == null) {
+                    return true
+                }
+            }
+
+            // check if a class is noted as abstract in the text
+            return psiClass.text.replace(" ", "")
+                .contains("abstractclass${psiClass.name}", ignoreCase = true)
+        }
+
     /**
      * Checks if the constraints on the selected class are satisfied, so that EvoSuite can generate tests for it.
      * Namely, it is not an enum and not an anonymous inner class.
      *
-     * @param psiClass the PSI class of interest
      * @return true if the constraints are satisfied, false otherwise
      */
-    private fun validateClass(psiClass: PsiClass): Boolean {
+    fun isValid(): Boolean {
         return !psiClass.isEnum && psiClass !is PsiAnonymousClass
     }
+}
 
-    /**
-     * Checks if the selected line contains (part of) a statement and is a valid line to be selected for EvoSuite.
-     * Namely, the line is within a method, it is not blank and contains (part of) a statement.
-     *
-     * @param selectedLine selected line number
-     * @param psiMethod surrounding PSI method
-     * @param psiFile containing PSI file
-     * @return true if the line is valid, false otherwise
-     */
-    private fun validateLine(
-        selectedLine: Int,
-        psiMethod: PsiMethod,
-        psiFile: PsiFile,
-    ): Boolean {
-        val doc: Document = PsiDocumentManager.getInstance(psiFile.project).getDocument(psiFile) ?: return false
+class JavaPsiHelper(private val psiFile: PsiFile) : PsiHelper {
 
-        val psiMethodBody: PsiCodeBlock = psiMethod.body ?: return false
-        if (psiMethodBody.statements.isEmpty()) return false
+    override val language: String
+        get() = "Java"
```

Comment:
How do you find an idea to introduce an enum with a set of languages (Java, Kotlin for now) that would contain their string representation:
```kotlin
enum class Language(val name) {
  Java("Java"), Kotlin("Kotlin");
}
```
And then use it here.